modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/GridDistributedCacheAdapter.java

```diff
@@ -164,6 +169,9 @@ public void removeVersionedEntry(KeyCacheObject key, GridCacheVersion ver) {
 
     /** {@inheritDoc} */
     @Override public void removeAll() throws IgniteCheckedException {
+        if (ctx.transactional() && ctx.grid().transactions().tx() != null)
```

Comment:
Actually there is no need to check whether cache is transactional. For atomic caches and transactions there is already check in `GridCacheGateway#checkAtomicOpsInTx`. Let's remove this condition.

Reply:
For clear operations - too